src/hubbleds/components/spectrum_measurement_tutorial_sequence/spectrum_measurement_tutorial.py

```diff
@@ -169,7 +180,14 @@ def __init__(self, viewer_layouts, *args, **kwargs):
                     deactivate_cb = partial(self.toggle_tower_select,self.dotplot_viewer_2), 
                     deactivate_before_tool=True)
         
-            
+        # run through steps to open run necessary setup calls in the vue file
+        for i in range(self.maxStepCompleted):
+            self.step = i
+    
+    def _on_tutorial_state_change(self, change):
+        self.print_log(f'name: {change["name"]}, old: {change["old"]}, new: {change["new"]}')
+        self.saving_state.update({change['name']:change['new']})
+        self.tutorial_state = self.saving_state.copy()
```

Comment:
Are `tutorial_state` and `saving_state` ever used differently? I guess I'm not sure why there are two dictionaries with the same content.

Reply:
I did this because otherwise it seemed like the corresponding `spectrum_tut_state` in `stage_1.py`wasn't realizing that it had been changed somewhere else. It worked with this approach, so this just seemed safer at the risk of perhaps _poor_ style or extra steps.